File: frontend/src/api/endpoints/devSettings.ts
import { apiFetch } from "../client";

export interface DevSettings {
  dev: boolean;
  stub_mode: boolean;
  stub_fail: boolean;
}

export const devSettingsApi = {
  get: () => apiFetch<DevSettings>("/api/v2/dev-settings"),
  update: (settings: Pick<DevSettings, "stub_mode" | "stub_fail">) =>
    apiFetch<DevSettings>("/api/v2/dev-settings", {
      method: "PATCH",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(settings),
    }),
};

File: backend/api/v2/endpoints/dev_settings.py
from fastapi import APIRouter, HTTPException, Request
from pydantic import BaseModel, ConfigDict, StrictBool

router = APIRouter(tags=["development"])


class DevSettingsUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")
    stub_mode: StrictBool
    stub_fail: StrictBool


@router.get("/api/v2/dev-settings")
def get_dev_settings(request: Request):
    return request.app.state.dev_settings.snapshot()


@router.patch("/api/v2/dev-settings")
def update_dev_settings(payload: DevSettingsUpdate, request: Request):
    settings = request.app.state.dev_settings
    if not settings.dev:
        raise HTTPException(status_code=404, detail="Not found")
    return settings.update(**payload.model_dump())
